services/src/main/java/org/keycloak/services/managers/RealmManager.java

```diff
@@ -406,6 +406,7 @@ private void setupAccountManagement(RealmModel realm) {
             client = KeycloakModelUtils.createClient(realm, Constants.ACCOUNT_MANAGEMENT_CLIENT_ID);
             client.setName("${client_" + Constants.ACCOUNT_MANAGEMENT_CLIENT_ID + "}");
             client.setEnabled(true);
+            client.setPublicClient(true);
```

Comment:
Just a suggestion (didn't do a full review). Couldn't we do this only when account2 profile is enabled?